### PR TITLE
Added some methods to quaternion

### PR DESCRIPTION
**Pull request by tomasmahlberg**
`toAxisAngle`, `toEulerVector` and `power`.

### CODE REVIEW

**Comment by tomasmahlberg** [comment]
@marcusnaslund review

File: source/geometry/Quaternion.ooc [hunk 1]
```diff
@@ -165,6 +166,16 @@ Quaternion: cover {
 			0.0f
 		)
 	}
+	toAxisAngle: func -> (FloatVector3D, Float) {
+		angle := 2.f * this real acos()
+		factor := (1.f - this real squared) sqrt()
+		axis := factor equals(0.f) ? FloatPoint3D new(1.f, 0.f, 0.f) : this imaginary / factor
+		(FloatVector3D new(axis x, axis y, axis z), angle)
+	}
+	toEulerVector: func -> FloatVector3D {
```
Comment (marcusnaslund) [inline source/geometry/Quaternion.ooc]: Can you squeeze a test of this in somewhere too? :)

**Comment by tomasmahlberg** [comment]
Updated.

**Comment by marcusnaslund** [comment]
Thanks.

**Pull request merged**